### DISCUSSION TITLE
Adjust the way the osu!taiko editor handles object placement

### DISCUSSION THREAD

**Discussion by Hiviexd** [ppy/osu#25150] [Ideas] [answered]
something negative that me (and others) took notice on across the years is how the taiko editor handles object placement, it's basically freeform placement across the playfield.

this works for i.e. the mania editor because the snaps are shown on the playfield, this does not currently work for taiko because you're placing notes on a blank slate, which forces you to keep looking at the upper timeline while placing notes, which leads to unwanted and annoying usage of peripheral vision to accomplish a core task.

to compensate for this while mapping, i found myself keeping the mouse fixed on the hit area when placing notes, which isn't optimal because i have to re-adjust my positioning every short while.

for an infinitely better approach UX-wise, i propose one of the following:

1. show snaps on the playfield
or
2. lock note placement to the hit area/current time

both options accomplish the goal of focusing on 1 main area when doing a core mapping task, however, I'm much more in favor of option 1 because you get to keep all the intended benefits of having a live playfield while having a nice UX for taiko mapping. i only mentioned option 2 because it has been suggested to me a while ago and its benefit is having a more faithful "classic" experience where mappers would have the upper timeline as their main focus. i still think option 1 is superior tho.

**Comment by peppy** [comment]
I am not sure why you're mentioning mania here as it *does* already show a snap grid.

We can easily move this over to taiko (it was already used in the catch editor as well).

**Reply by peppy** [reply, marked as answer]
https://github.com/ppy/osu/pull/25154

**Reply by Hiviexd** [reply]
FWIW the mania mention was a result of a misunderstanding from my part, apologies. I've removed the mention.
